linera-views-derive/src/lib.rs

```diff
@@ -90,18 +61,14 @@ fn get_extended_entry(e: Type) -> TokenStream2 {
 }
 
 fn generate_view_code(input: ItemStruct, root: bool) -> TokenStream2 {
+    let template_vect = get_seq_parameter(input.generics.clone());
+    let (context, context_constraints) = context_and_constraints(&input, &template_vect);
     let struct_name = input.ident;
     let (impl_generics, type_generics, maybe_where_clause) = input.generics.split_for_impl();
-    let template_vect = get_seq_parameter(input.generics.clone());
-
-    let (context, context_constraints) = context_and_constraints(&input.attrs, &template_vect);
-
-    let mut where_clause = maybe_where_clause
-        .cloned()
-        .unwrap_or_else(empty_where_clause);
-    where_clause
-        .predicates
-        .extend(context_constraints.predicates);
+    let input_constraints = maybe_where_clause
+        .map(|w| w.predicates.iter().cloned())
+        .into_iter()
+        .flatten();
```

Comment:
That code section is duplicated in `generate_view_code`, `generate_hash_view_code` and in general all the function generating code.
So, I think it would make sense to add it to the `context_and_constraints`

Reply:
https://github.com/linera-io/linera-protocol/pull/3854/commits/3419ca31d54572b0fe8a956f0a624a8f9bdd2e85